### PR TITLE
[Quant] Add fused conv2d_add op for onednn backend

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #91155
* #91154
* #91153
* #91152
* #90818
* #90364
* __->__ #90262

**Summary**
Post op fusion can reduce data movement overhead and improve inference performance. This PR adds fused `conv2d_add` op for onednn backend, which will be used for int8 inference with onednn backend. Cannot call this op with other quantization backends otherwise an error is thrown.

**Test Plan**
```
python -m pytest test_quantization.py::TestQuantizedConv
```

cc @VitalyFedyunin @jgong5 @mingfeima @XiaobingSuper @sanchitintel @ashokei @jingxu10 @gujinghui @PenghuiCheng @jianyuh @min-jean-cho @yanbing-j @Guobing-Chen @Xia-Weiwen

### CODE REVIEW

**Review by jgong5 — COMMENTED** [review]
What's the plan of supporting this on x86 backend? Also, why does it require ideep upgrade?

**Comment by leslie-fang-intel** [comment]
> What's the plan of supporting this on x86 backend? 

We plan to enable it on `oneDNN backend` firstly to verify the performance and porting it to `x86 backend` when test results shows this fusion with `oneDNN backend` has better performance than `FBGEMM backend`. Then in `x86 backend`, we will let this fusion go through oneDNN implementation.

> Also, why does it require ideep upgrade?

Current ideep doesn't support post-op sum fusion with zero point. I have enabled it on a ideep PR: https://github.com/intel/ideep/pull/148. @jgong5 Please also help to take a look.

**Review by jgong5 — APPROVED** [review]
LGTM. I added some comment on the ideep PR.

**Comment by jerryzh168** [comment]
please put ideep third-party update in a separate PR as well

**Comment by leslie-fang-intel** [comment]
> please put ideep third-party update in a separate PR as well

Thanks, I will put ideep update in a separate PR when the ideep changes ready in its master.

**Comment by leslie-fang-intel** [comment]
The change of ideep has been moved to a separate PR.

File: test/quantization/core/test_quantized_op.py [hunk 1]
```diff
@@ -4687,7 +4709,7 @@ def _test_qconv_impl(
            Y_scale=st.floats(4.2, 5.6),
            Y_zero_point=st.integers(0, 4),
            use_bias=st.booleans(),
-           use_relu=st.booleans(),
+           post_op=st.sampled_from(["none", "relu"]),
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_op.py:4712]: can be a separate PR, but might make sense to split the conv and conv_relu test as well

File: test/quantization/core/test_quantized_op.py [hunk 2]
```diff
@@ -4780,7 +4886,7 @@ def test_qconv2d(
            Y_scale=st.floats(4.2, 5.6),
            Y_zero_point=st.sampled_from([0]),
            use_bias=st.booleans(),
-           use_relu=st.booleans(),
+           post_op=st.sampled_from(["none", "relu"]),
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_op.py:4889]: same here

**Review by jerryzh168 — APPROVED** [review]

File: test/quantization/core/test_quantized_op.py [hunk 3]
```diff
@@ -4750,7 +4772,91 @@ def test_qconv2d(
...
+            input_channels = input_channels_per_group * groups
+            output_channels = output_channels_per_group * groups
+            kernels = (kernel_h, kernel_w)
+            strides = (stride_h, stride_w)
+            pads = (pad_h, pad_w)
+            dilations = (dilation, dilation)
+
+            qconv = torch.ops.quantized.conv2d
+            if post_op == "add":
+                qconv = torch.ops.quantized.conv2d_add
```
Comment (jerryzh168) [inline test/quantization/core/test_quantized_op.py:4840]: if this is only "add" we can remove the post_op argument

File: test/quantization/core/test_quantized_op.py (hunk 1 above)
Comment (leslie-fang-intel) [reply test/quantization/core/test_quantized_op.py]: Thanks for the suggestions, I have split the conv and conv_relu test and so as the other similar test cases.

File: test/quantization/core/test_quantized_op.py (hunk 2 above)
Comment (leslie-fang-intel) [reply test/quantization/core/test_quantized_op.py]: Thanks for the suggestions, split them into separate tests.

File: test/quantization/core/test_quantized_op.py (hunk 3 above)
Comment (leslie-fang-intel) [reply test/quantization/core/test_quantized_op.py]: Thanks for the suggestions, I have removed the post_op argument and the check. In next PR, I will put conv2d_add_relu into   a separate test.
